cmake/modules/HCT.cmake

```diff
@@ -44,7 +44,7 @@ function(add_hlsl_hctgen mode)
     message(FATAL_ERROR "add_hlsl_hctgen requires OUTPUT argument")
   endif()
  
-  set(temp_output ${CMAKE_CURRENT_BINARY_DIR}/${ARG_OUTPUT}.tmp)
+  set(temp_output ${CMAKE_CURRENT_BINARY_DIR}/tmp/${ARG_OUTPUT})
```

Comment:
This assumes hctgen creates directories in its output paths if they don't exist. I think that's the case so no need to change it if so, but I've seen this kind of thing trip tools up before so worth double checking.

Reply:
Where were you last year when I made that mistake in the first place? It was fixed in #4571 😁